### PR TITLE
add post return, add the URI to header location.

### PR DESCRIPTION
**Pull request by wy65701436**
Add a comment function in base to handle the post method return resource uri.

Passed on the API and UI testing.

### CODE REVIEW

File: api/project.go [hunk 1]
```diff
@@ -87,11 +92,14 @@ func (p *ProjectAPI) Post() {
...
 	}
 	project := models.Project{OwnerID: p.userID, Name: projectName, CreationTime: time.Now(), Public: public}
-	err = dao.AddProject(project)
+	projectID, err := dao.AddProject(project)
 	if err != nil {
 		log.Errorf("Failed to add project, error: %v", err)
 		p.RenderError(http.StatusInternalServerError, "Failed to add project")
 	}
+
+	projectLocation := p.HarborURL + "/api/projects/" + strconv.FormatInt(projectID, 10)
```
Comment (reasonerjt) [inline api/project.go]: Is it possible to get url information in the http request?

Comment (wy65701436) [inline api/project.go]: good catch, will have a update on this.

**Comment by reasonerjt** [comment]
LGTM, will you update the swagger accordingly?

**Comment by wy65701436** [comment]
sure, will have a batch update on the recent code change...

**Pull request merged**